### ISSUE TITLE
CoinbasePro: watchOrderBook trigger frequency out of sync with exchange

### ISSUE THREAD

**Issue by xc5211** [ccxt/ccxt#16207]
While cctx wiki says `In real-time mode CCXT Pro will return the result as soon as each new delta arrives from the exchange.`, I've monitored that for CoinbasePro(not sure if other exchange having this issue or not), watchOrderBook would only be triggered after multiple batches of deltas arrive from the exchange. I'm wondering if this is expected or things need to be fixed here?

**Comment by frosty00** [comment]
what language and what code are you using?

**Comment by pcriadoperez** [comment]
I confirmed this issue. @frosty00 , could this be due to the issue you mentioned of subscriptions getting overwritten when messages come in very fast?

**Comment by ttodua** [comment]
I'll close this topic, because `coinbasepro` is no longer a primary class for `coinbase` integration (users are advised to migrate & use regular `coinbase` class. 
the out-of-sync issue might be related to the missed updates (as mentioned in above comment by Pablo), so as we are going to release event-driver subscriptions soon, that will be another nuance to solve that even if users use `coinbasepro` class.

please let us know if still any issues

**Comment by xc5211** [comment]
Is there a ticket that I can follow the progress of the event-driven subscription? @ttodua

**Comment by ttodua** [comment]
@xc5211 yes, either announcement channel: https://docs.ccxt.com/#/README?id=announcements
or https://github.com/ccxt/ccxt/pull/21010
